### DISCUSSION TITLE
https://anonymity-0.github.io/posts/%E7%BD%91%E7%BB%9C%E5%AE%89%E5%85%A8%E4%BD%9C%E4%B8%9A5/

### DISCUSSION THREAD

**Discussion by giscus** [Anonymity-0/Anonymity-0.github.io#7] [Announcements]
# https://anonymity-0.github.io/posts/%E7%BD%91%E7%BB%9C%E5%AE%89%E5%85%A8%E4%BD%9C%E4%B8%9A5/

得分：10/10
C程序homework08.c的主函数如下：
1 2 3 4 5 6 7 8 9 10 11 12 int main(int argc, char * argv[]) { init_buf(Lbuffer, LEN); switch(argc) { case 1: f00(); break; case 2: f01(); break; case 3: f02(); break; default: f00(); break; } puts(&#34;Done.\nThe program exited normally.&#34;); return 0; } 在32位的ubuntu16.04系统中用gcc -fno-stack-protector 编译该程序，得到的可执行程序见附件，通过gdb调试，对f00()、f01()和f02()进行分析： (1) 函数f00()、f01()和f02()是否导致段错误。 如果没有命令行参数（argc 等于 1，因为程序名本身也算一个参数），则执行函数 f00()。 如果有一个命令行参数（argc 等于 2），则执行函数 f01()。 如果有两个命令行参数（argc 等于 3），则执行函数 f02()。 如果有三个或更多的命令行参数，还是执行函数 f00()。 ![[Pasted image 20231122231415.png]] 函数foo(), 和 foo01() 都会导致段错误。

https://anonymity-0.github.io/posts/%E7%BD%91%E7%BB%9C%E5%AE%89%E5%85%A8%E4%BD%9C%E4%B8%9A5/

**Comment by CZXczx666** [comment]
谢谢学长
